agent/config.go

```diff
@@ -2121,6 +2123,99 @@ func ReadConfigPaths(paths []string) (*Config, error) {
 	return result, nil
 }
 
+// ResolveTmplAddrs iterates over the myriad of addresses in the agent's config
+// and performs go-sockaddr/template Parse on each known address in case the
+// user specified a template config for any of their values.
+func (c *Config) ResolveTmplAddrs() (err error) {
+	parse := func(addr *string, name string) {
+		if *addr == "" || err != nil {
+			return
+		}
+		var ip string
+		ip, err = parseSingleIPTemplate(*addr)
+		if err != nil {
+			err = fmt.Errorf("Resolution of %s failed: %v", name, err)
+			return
+		}
+		if strings.HasPrefix(ip, "unix://") {
```

Comment:
this introduces a small difference in behavior that I want to point out. Previously if you set a unix:// address for AdvertiseAddr or AdvertiseAddrWAN you'd get an error immediately when it net.Parse fails, but now you won't anymore. It might fail further downstream (need to test that out).  It's probably okay to fail later than at this method, but need to confirm that.